src/lib/Microsoft.Health.Fhir.Ingest.Template/MeasurementExtractor.cs

```diff
@@ -77,24 +78,26 @@ protected T EvalExpression<T>(JToken token, string name, bool isRequired = false
                         }
                         catch (Exception e)
                         {
-                            exceptions.Add(new IncompatibleDataException($"Encounted an error while extracting value for [{name}] using expression {expression.Value}", e));
+                            exceptions.Add(new IncompatibleDataException($"Encounted an error while extracting value for [{name}] using expression {expression.Value}", e, valueLineInfo));
                         }
                     }
                 }
 
                 if (isRequired)
                 {
+                    var lineInfo = expressions.First(e => e != null);
```

Comment:
First() will throw an exception if there aren't elements that satisfy the criteria.  I don't think this is what you want given the checks you have below.  You can use FirstOrDefault() and the lineInfo expression will be null if nothing is found.

If you do want the First() call to throw if nothing is found it will be an invalid operation exception, https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.first?view=net-6.0.  You will want to make sure the code up the stack is handling it properly.

Reply:
Good point. The check on line 48 ensures that I have at least one non-null `TemplateExpression`. So this would never throw. But looking at it more, I think I'll refactor this whole method to take in a single `TemplateExpression` versus an var-args list of expressions. That will simplify the logic here.

Reply:
Actually nevermind; it looks like we do require the use of var-ags list.